node/src/accountant/scanners_tools.rs

```diff
@@ -531,9 +518,52 @@ mod tests {
             response_skeleton_opt: None,
         };
 
-        let (ok, err) = separate_early_errors(&sent_payable, &Logger::new("test"));
+        let (ok, err) = separate_early_errors(&sent_payable, &Logger::new(test_name));
 
         assert_eq!(ok, vec![payable_ok]);
-        assert_eq!(err, vec![error])
+        assert_eq!(err, vec![error.clone()]);
+        TestLogHandler::new().exists_log_containing(&format!(
+            "WARN: {}: Outbound transaction failure due to '{:?}",
+            test_name, error
+        ));
+    }
+
+    #[test]
+    #[should_panic(expected = "Unable to calculate elapsed time for the scan.")]
+    fn update_timestamp_and_log_panics_if_timestamp_is_wrong() {
+        let time_in_future = SystemTime::now().add(Duration::seconds(10));
+
+        update_timestamp_and_log(
+            &mut Some(time_in_future),
+            ScanType::Payables,
+            &Logger::new("test"),
+        );
+    }
+
+    #[test]
+    fn update_timestamp_and_log_if_timestamp_is_correct() {
+        init_test_logging();
+        let test_name = "update_timestamp_and_log_if_timestamp_is_correct";
+        let time_in_past = SystemTime::now().sub(Duration::seconds(10));
+        let logger = Logger::new(test_name);
+
+        update_timestamp_and_log(&mut Some(time_in_past), ScanType::Payables, &logger);
+
+        TestLogHandler::new().exists_log_matching(&format!(
+            "INFO: {test_name}: The Payables scan ended in \\d+ms."
```

Comment:
The `time` is tested in other tests. Here's one example which accurately tests the time `payable_scanner_can_initiate_a_scan`. Hence, I don't feel the need to add an approximate assertion to this test.

In the end I believe it's better to limit the test to assertions for which it was written on the first place, instead of making it over complicated by adding tighter assertions.

Reply:
The closer the assertion is to where the new timestamp is generated the better, probably. Yeah, no big deal...
...mostly because, as I noticed this test is gone...

However it makes me sad that I missed that this test was just half done. 
I were testing the printed message but what you maybe should have test in the first place was what happened to the value of 'time_in_past`. An assertion on the value was obviously missing. 

Printing message does not mean the same as setting a state of an outer value. So you didn't provide a view to understand what happened inside the black box, that function. The test was saying only printing a message was going on.